libbeat/processors/add_docker_metadata/add_docker_metadata.go

```diff
@@ -63,8 +67,30 @@ type addDockerMetadata struct {
 	dedot     bool          // If set to true, replace dots in labels with `_`.
 }
 
+// checkForDockerEnvironment checks if Docker is available in running environment
+// and returns an error in case Docker is not detected
+func checkForDockerEnvironment() error {
+	cli, err := client.NewEnvClient()
+	errorMsg := fmt.Sprintf("%v: docker environment not detected.", processorName)
+	if err != nil {
+		logp.Info(errorMsg)
+		return fmt.Errorf(errorMsg)
+	}
+	info, err := cli.Info(context.Background())
+	if err != nil {
+		logp.Info(errorMsg)
+		return fmt.Errorf(errorMsg)
+	}
+	logp.Info("docker environment detected: %v", info)
+	return nil
+}
+
 // New constructs a new add_docker_metadata processor.
 func New(cfg *common.Config) (processors.Processor, error) {
+	err := checkForDockerEnvironment()
+	if err != nil {
+		return nil, nil
```

Comment:
Is it ok to return a `nil` processor here? Will something try to call `Run` over this nill object?

Reply:
This is right. I should not return nil here. Thank you, will be addressed!